src/execution_plan/ops/shared/update_functions.c

```diff
@@ -57,28 +59,21 @@ void CommitUpdates
 
 	dictEntry *entry;
 	dictIterator *it = HashTableGetIterator(updates);
-	MATRIX_POLICY policy = Graph_GetMatrixPolicy(gc->g);
-	Graph_SetMatrixPolicy(gc->g, SYNC_POLICY_NOP);
+
+	if(type == ENTITY_NODE) {
+		UpdateLabels(gc, add_labels, remove_labels, true);
+	}
 
 	while((entry = HashTableNext(it)) != NULL) {
 		PendingUpdateCtx *update = HashTableGetVal(entry);
 
-		// if entity has been deleted, perform no updates
-		if(GraphEntity_IsDeleted(update->ge)) continue;
```

Comment:
Why was this line removed?

Reply:
no needed anymore now that the delete operation is eager the check in EvalEntityUpdates